ports/rp2/mpnetworkport.c

```diff
@@ -67,8 +69,13 @@ void cyw43_irq_init(void) {
     NVIC_SetPriority(PendSV_IRQn, PICO_LOWEST_IRQ_PRIORITY);
 }
 
+int cyw43_has_pending(void) {
+    return has_pending;
+}
+
 void cyw43_post_poll_hook(void) {
     gpio_set_irq_enabled(CYW43_PIN_WL_HOST_WAKE, CYW43_IRQ_LEVEL, true);
+    has_pending = 0;
```

Comment:
I think this line needs to come before reenabling the GPIO IRQ.  Otherwise there's a race condition, the IRQ could fire just before setting `has_pending` to 0.

Reply:
Yes, this seems reasonable. And additionally as this is executed with pendsv-priority, from the view of other code (that might want to go into light sleep or similar), this seems atomic so changing it doesn't hurt semantics.